AutoGraph/WebSocketClient.swift

```diff
@@ -237,6 +238,33 @@ open class WebSocketClient {
         self.queuedSubscriptions.removeAll()
         self.attemptReconnectCount = kAttemptReconnectCount
         self.state = .disconnected
+        self.isConnectionAck = false
+    }
+    
+    func writeQueuedSubscriptions() {
+        let queuedSubscriptions = self.queuedSubscriptions
+        self.queuedSubscriptions.removeAll()
+        queuedSubscriptions.forEach { (_, connected: WebSocketConnected) in
+            connected(.success(()))
+        }
+    }
+    
+    func resendSubscriptionIfNoConnectionAck() {
+        self.connectionAckWorkItem?.cancel()
+        self.connectionAckWorkItem = nil
+        
+        guard self.isConnectionAck == false else {
+            return
+        }
+        
+        let connectionAckWorkItem = DispatchWorkItem { [weak self] in
+            self?.requeueAllSubscribers()
+            self?.disconnectAndPossiblyReconnect()
+            self?.webSocket.connect()
```

Comment:
this is the same stuff as in `reconnect` right? can factor into a `private performReconnect` or something? (failing to come up with a better name)

Reply:
good call